rift/api/resources.py

```diff
@@ -154,6 +158,57 @@ def on_delete(self, req, resp, tenant_id, target_id):
         Target.delete_target(target_id=target_id)
 
 
+class PingTargetResource(ApiResource):
+
+    def on_get(self, req, resp, tenant_id, target_id):
+        target = Target.get_target(tenant_id, target_id)
+        if target:
+            # Nova
+            if 'nova' in target.address.as_dict().keys():
```

Comment:
You use target.address.* quite a bit, so I would assign that to a local variable, to keep from deep referencing a lot.